controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -795,6 +797,80 @@ func TestRebootWithSuffixedAnnotation(t *testing.T) {
 	)
 }
 
+// TestRebootWithServicing tests full reboot cycle with suffixless
+// annotation and servicing.
+func TestRebootWithServicing(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = make(map[string]string)
+	host.Annotations[metal3api.RebootAnnotationPrefix] = ""
+	host.Status.PoweredOn = true
+	host.Status.Provisioning.State = metal3api.StateProvisioned
+	host.Spec.Online = true
+	host.Spec.Image = &metal3api.Image{URL: "foo", Checksum: "123"}
+	host.Spec.Image.URL = "foo"
+	host.Spec.Firmware = &metal3api.FirmwareConfig{
+		VirtualizationEnabled: ptr.To(true),
+	}
+	host.Status.Provisioning.Image.URL = "foo"
+	hup := &metal3api.HostUpdatePolicy{
+			TypeMeta: metav1.TypeMeta{
+			Kind: "HostUpdatePolicy",
+			APIVersion: "metal3.io/v1alpha1",
+		},
+			ObjectMeta: metav1.ObjectMeta{
+			Name:      host.Name,
+			Namespace: namespace,
+		},
+		Spec: metal3api.HostUpdatePolicySpec{
+			FirmwareSettings: metal3api.HostUpdatePolicyOnReboot,
+			FirmwareUpdates: metal3api.HostUpdatePolicyOnReboot,
+		},
+	}
+	// hup.ResourceVersion = ""
+	r := newTestReconciler(host, hup)
+	err := r.Update(context.TODO(), hup)
```

Comment:
nit: I'm not sure you need to explicit create the object after adding it to the test reconciler.